docs/hugo/content/contributing/add-a-new-code-generated-resource/write-a-test.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Write a test for the resource
+linktitle: Write a test
+weight: 40
+---
+
+Every new resource should have a handwritten test as there is always the possibility of a resource behaving in a subtly different way than expected. This also applies to _new versions_ of existing resources.
+
+All these tests live in the [`v2/internal/controllers`](https://github.com/Azure/azure-service-operator/tree/main/v2/internal/controllers). There's a readme in that folder that describes the test structure and naming conventions.
+
+For each test there's a recording in the `recordings` folder that records the HTTP interactions between the test and Azure allowing for rapid replay for future runs.
+
+One good approach is to start from an [existing test](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/documentdb_mongodb_crud_v20231115_test.go) and modify it to work for your resource. It can also be helpful to refer to examples in the [ARM templates GitHub repo](https://github.com/Azure/azure-quickstart-templates).
```

Comment:
minor: Maybe worth calling out the philosophy of doing as much in parallel as possible in the test, for example in the test you link:
```
	// Create both resources in Azure
	tc.CreateResourcesAndWait(&acct, &db)
```

and the user of `RunParallelSubtests`, to simulate what happens in reality (user applies YAML with everything).